### PR TITLE
Refactor profile page

### LINKED ISSUE #2077: suggestion: profiles should have the standard activity feed on them, instead of trying to parse everything out into 'completed bounties' and 'started bounties' etc
example https://gitcoin.co/profile/PumpkingWok?foo=bar

suggestion: profiles should have the standard activity feed on them, instead of trying to parse everything out into 'completed bounties', tips received, and 'started bounties' etc 

this would be much simpler from an implementation standpoint, and also more represents how users will visually look at the profile

### PR DESCRIPTION
**Pull request by SaptakS**
##### Description
Refactor profile page to show all activities in tabular format.

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Refers/Fixes

Fixes #2077

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@SaptakS good for review?

**Comment by SaptakS** [comment]
@thelostone-mc yes!! https://stage.gitcoin.co/profile/thelostone-mc
Though I need some opinion on this from @owocki @PixelantDesign and @mbeacom. Actually 2 things.

1. I am showing Tasks Completed, Bounties Funded and Tips Received. Why not show Submitted Work and Work Started also?

2. I am showing all the activities. This mean if an user has submitted work and work is accepted, both will show up in All Activities. Is that what we want?

**Comment by thelostone-mc** [comment]
@SaptakS added a few media queries and stuff !

File: app/dashboard/templates/profiles/profile.html [hunk 1]
```diff
@@ -119,203 +120,118 @@
...
-                        <div class="profile-bounties__group-title profile-bounties__group-title--completed">
-                          <i class="far fa-check-circle"></i>
-                          <span class="pl-1">
-                            {% trans "Completed" %} {{ group.completed|length }}
-                            Bount{{ group.completed|pluralize:"y,ies" }}
-                          </span>
+
+                {% if group.activity_bounties %}
+                <div id="activity-tabs" class="tab-container mt-4 font-body">
+                  <button type="button" id="all" class="section-tab">All Activity ({{ group.activity_bounties|length }})</button>
```
Comment (mbeacom) [inline app/dashboard/templates/profiles/profile.html]: You can use `{{ group.activity_bounties.count }}`

File: app/dashboard/templates/profiles/profile.html [hunk 2]
```diff
@@ -119,203 +120,118 @@
...
+                  <button type="button" id="funded" class="section-tab">Bounties Funded ({{ funded_bounties_count }})</button>
+                  <button type="button" id="completed" class="section-tab">Bounties Completed</button>
```
Comment (mbeacom) [inline app/dashboard/templates/profiles/profile.html]: Can we wrap the copy in `trans` throughout?

File: app/dashboard/models.py [hunk 3]
```diff
@@ -2198,21 +2198,14 @@ def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
         }
 
         if activities:
-            fulfilled = self.fulfilled.filter(
+            all_activities = self.activities.filter(
                 bounty__network=network
```
Comment (mbeacom) [inline app/dashboard/models.py]: Can we pipe this together with `self.activites.filter(tip__network=network)` or combine with `Q()` ?

File: app/dashboard/templates/shared/profile_activities_bounty.html [hunk 4]
```diff
@@ -8,7 +8,7 @@
   <div class="col-12 col-md-8 d-flex flex-column">
     <div class="bounty-detail">
       <div class="title font-subheader">
-        {{ bounty.title }}
+        {{ bounty.title}}
```
Comment (mbeacom) [inline app/dashboard/templates/shared/profile_activities_bounty.html]: 😭

File: app/dashboard/templates/profiles/profile.html (hunk 1 above)
Comment (mbeacom) [reply app/dashboard/templates/profiles/profile.html]: Actually, disregard the `count`, since we're evaluating the QS below anyway.

**Comment by owocki** [comment]
does the stage profile mentioned still have this code running on it? https://stage.gitcoin.co/profile/thelostone-mc

**Comment by owocki** [comment]
@SaptakS OMG I `<3` this !  Didnt check data for accuracy, but it *looks * great... Once we get Kudos launched hopefully we can get into this format.  Lets get it shipped soon!

**Comment by SaptakS** [comment]
@owocki my only questions are the ones mentioned [here](https://github.com/gitcoinco/web/pull/2237#issuecomment-422249471). Apart from that, it is good to be reviewed and merged.
Also, I discussed with @octavioamu, so getting the kudos sections integrated should be as straightforward as including a html template in jinja2.

**Comment by SaptakS** [comment]
@owocki I can definitely do that addition of dates. Will update with the date and the other two tabs.

**Comment by mbeacom** [comment]
_puts on scope creep hat_
Should we have some kind of `activity type` visual identifier on the activity elements?

**Comment by owocki** [comment]
> puts on scope creep hat

its not "scope creep", its "progressive elaboration" :) 

> Should we have some kind of activity type visual identifier on the activity elements?

we have one of these on `/activity` -- possible to just reuse that?

**Comment by SaptakS** [comment]
Will check. Thanks for the feedback. @mbeacom you look great with any hat you put... 😛

**Comment by SaptakS** [comment]
@owocki @mbeacom done! Ready for review.

**Comment by PixelantDesign** [comment]
yay!

File: app/dashboard/templates/shared/profile_activities_bounty.html [hunk 5]
```diff
@@ -1,45 +1,64 @@
...
-            {{ bounty.experience_level }} &bull;
+          {% if activity.bounty.experience_level %}
+            {{ activity.bounty.experience_level }} &bull;
           {% endif %}
-          {% if bounty.status %}
-            {{ bounty.status|title }}
+          {% if activity.bounty.status %}
+            {{ activity.bounty.status|title }}
           {% endif %}
+          {% if activity.activity_type == 'new_bounty' %}
```
Comment (mbeacom) [inline app/dashboard/templates/shared/profile_activities_bounty.html]: I'm still interested to see how this performs on live with fairly populated activities 🤔I'm fine with it, but we might need to patch it depending on the performance.

File: app/dashboard/models.py [hunk 6]
```diff
@@ -2198,21 +2198,14 @@ def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
         }
 
         if activities:
-            fulfilled = self.fulfilled.filter(
+            all_activities = self.activities.filter(
```
Comment (mbeacom) [inline app/dashboard/models.py]: Did you decide to exclude tips for now?

**Comment by owocki** [comment]
this looks so much better!  can this go out soon!?

**Comment by thelostone-mc** [comment]
@mbeacom I believe we can exclude tips 🤔  We good with this ?

File: app/dashboard/templates/shared/profile_activities_bounty.html (hunk 5 above)
Comment (SaptakS) [reply app/dashboard/templates/shared/profile_activities_bounty.html]: Yes. I agree to that. I would like to see the same.

File: app/dashboard/models.py (hunk 6 above)
Comment (SaptakS) [reply app/dashboard/models.py]: The tips we are already sending in a non activity way. So i am excluding it in the activities.

**Comment by owocki** [comment]
new profile is looking hot!

@vs77bb and i just posted a few cleanup items from this PR over at https://github.com/gitcoinco/web/issues

**Pull request merged**